dali/kernels/signal/resampling.h

```diff
@@ -132,11 +181,27 @@ struct Resampler {
         std::tie(i0, i1) = window.input_range(in_pos);
         if (i0 + in_block_i < 0)
           i0 = -in_block_i;
-        if (i1 + in_block_i >= n_in)
-          i1 = n_in - 1 - in_block_i;
+        if (i1 + in_block_i > n_in)
+          i1 = n_in - in_block_i;
         float f = 0;
         int i = i0;
 
+#ifdef __ARM_NEON
```

Comment:
Maybe we can extract this vectorized parts into separate functions, instead having a lot of variants for different architectures in one body.